169 add ad users to the db

Jasminmay16 opened the pull request:
### Overview
Completed the login functionality for the AD users. Now all the employees from the ad database have been added to our database. With Email and role
<img width="1108" height="616" alt="databaseimage" src="https://github.com/user-attachments/assets/b25adda9-96d4-4f3d-ada5-c21b7ff6ebcc" />

- When importing a user from Active Directory, the user's role is now fetched from the database and assigned to the UserRole property. Throws an exception if the role does not exist, ensuring data consistency.

### Problem
The only error is the user/employee Rasmus. As he doesn't have an email, which means we cannot delete him via the endpoints. It needs to be manually done through the database.

bukh04 commented on API/Controllers/HotelController.cs:
```diff
@@ -122,23 +109,55 @@ public async Task<ActionResult> PutHotel(HotelDto updateHotel)
         
     }
+    
+    [HttpPatch("{hotelId}")]
+    public async Task<IActionResult> PatchHotel(int hotelId, [FromBody]HotelPutDto updateDto)
+    {
+        var hotel = await _context.Hotels.FindAsync(hotelId);
+        if (hotel == null) return NotFound();
```
You have "PutHotel" which does the same thing, updates hotels. We use PutHotel on the admin panel, it works well

Jasminmay16 replied:
Yes! I forgot to remove this part. In my other PR with refactored endpoints, I have removed the PatchHotel, and just use the PutHotel :)

bukh04 commented on DomainModels/Dto/HotelDto.cs:
```diff
@@ -16,6 +16,7 @@ public class HotelDto
 
         [Required(ErrorMessage = "Address is required")]
 		public required string Address { get; set; }
+        public DateTime UpdatedAt { get; set; }
         public string? Description { get; set; }
```
Use this property then you update hotels

Jasminmay16 replied:
Apparently, there are unnecessary changes in the hoteldto and hotelcontroller files, I apologies. This PR is only meant for the AD, all the refactoring and refinement for endpoints are in another PR 😓.

Jasminmay16 commented:
Updated log messages in ActiveDirectoryService to use English and provide clearer information. Added logging of raw AD department values for debugging. Improved department normalization and hotel name matching logic for more robust mapping.

Jasminmay16 commented:
Fixed the ad user Rasmus :)
